### PR TITLE
Visit Table Paginator

### PR DESCRIPTION
**Pull request by Johngtka**
Added paginator to the visit table. Showing 10 records per slide

image

### CODE REVIEW

**Comment by Johngtka** [comment]
i used this example:
`https://material.angular.io/components/table/examples#table-overview`

**Comment by Johngtka** [comment]
S&M commit prolosal:
`feat: visit table paginator with 10 rows per page`

File: src/app/visit/visit.component.html [hunk 1]
```diff
@@ -10,61 +10,64 @@
...
+            </th>
+            <td mat-cell *matCellDef="let element">
+                {{ element.image }}
+            </td>
+        </ng-container>
+        <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+        <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
+    </table>
+    <mat-paginator
+        [length]="100"
```
Comment (macrog) [inline src/app/visit/visit.component.html:68]: `length` what is this? do we need it?

File: src/app/visit/visit.component.ts [hunk 2]
```diff
@@ -21,15 +29,15 @@ export class VisitComponent implements OnInit {
         'type',
         'image',
     ];
-    constructor(
-        private storeService: StoreService,
-        private snackService: SnackService,
-    ) {}
-
+    dataSource;
```
Comment (macrog) [inline src/app/visit/visit.component.ts:32]: do we need this? we had a `store` variable for that, if you want to use `dataSource` please remove `store`

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/visit/visit.component.html (hunk 1 above)
Comment (Johngtka) [reply src/app/visit/visit.component.html:68]: no and i remove it

Comment (Johngtka) [reply src/app/visit/visit.component.html:68]: obraz
this is a definition

File: src/app/visit/visit.component.ts [hunk 3]
```diff
@@ -21,15 +28,15 @@ export class VisitComponent implements OnInit {
...
```
Comment (macrog) [inline src/app/visit/visit.component.ts:31]: add type and model maybe?

File: src/app/visit/visit.component.ts [hunk 4]
```diff
@@ -21,15 +28,15 @@ export class VisitComponent implements OnInit {
...
     ngOnInit(): void {
         this.patient = history.state;
         this.storeService.getStore().subscribe({
-            next: (data) => (this.store = data.sort((a, b) => a.id - b.id)),
+            next: (data) => (
+                (this.dataSource = data.sort((a, b) => a.id - b.id)),
+                (this.dataSource = new MatTableDataSource<Store>(data)),
```
Comment (macrog) [inline src/app/visit/visit.component.ts:37]: this
`this.dataSource = new MatTableDataSource<Store>(data)`
should be:
`this.dataSource = new MatTableDataSource<Store>(this.dataSource)`

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/visit/visit.component.ts (hunk 3 above)
Comment (Johngtka) [reply src/app/visit/visit.component.ts:31]: what type must have dataSource?

Comment (macrog) [reply src/app/visit/visit.component.ts:31]: what data are you showing? `store`

**Review by macrog — APPROVED** [review]
works 👏 🚀

**Pull request merged**